bitnami/nessie/templates/_helpers.tpl

```diff
@@ -0,0 +1,323 @@
+      mountPath: {{ .Values.persistence.mountPath }}
+{{- end -}}
+
+{{- define "nessie.waitForDBInitContainer" -}}
+# We need to wait for the postgresql database to be ready in order to start with Nessie.
+- name: wait-for-db
+  image: {{ template "nessie.psql.image" . }}
+  imagePullPolicy: {{ .Values.psqlImage.pullPolicy }}
+  {{- if .Values.containerSecurityContext.enabled }}
+  securityContext: {{- include "common.compatibility.renderSecurityContext" (dict "secContext" .Values.containerSecurityContext "context" $) | nindent 4 }}
+  {{- end }}
+  command:
+    - bash
+    - -ec
+    - |
+      #!/bin/bash
+
+      set -o errexit
+      set -o nounset
+      set -o pipefail
+
+      . /opt/bitnami/scripts/liblog.sh
+      . /opt/bitnami/scripts/libvalidations.sh
+      . /opt/bitnami/scripts/libpostgresql.sh
+      . /opt/bitnami/scripts/postgresql-env.sh
+
+      {{- if .Values.usePasswordFile }}
+      export DATABASE_PASSWORD="$(< "/bitnami/nessie/secrets/database/QUARKUS_DATASOURCE_POSTGRESQL_PASSWORD")"
+      {{- end }}
+      info "Waiting for host $DATABASE_HOST"
+      psql_is_ready() {
+          if ! PGCONNECT_TIMEOUT="5" PGPASSWORD="$DATABASE_PASSWORD" psql -U "$DATABASE_USER" -d "$DATABASE_NAME" -h "$DATABASE_HOST" -p "$DATABASE_PORT_NUMBER" -c "SELECT 1"; then
+             return 1
+          fi
+          return 0
+      }
+      if ! retry_while "debug_execute psql_is_ready"; then
+          error "Database not ready"
+          exit 1
+      fi
+      info "Database is ready"
+  {{- if .Values.resources }}
+  resources: {{- toYaml .Values.resources | nindent 4 }}
+  {{- else if ne .Values.resourcesPreset "none" }}
+  resources: {{- include "common.resources.preset" (dict "type" .Values.resourcesPreset) | nindent 4 }}
+  {{- end }}
+  env:
+    - name: BITNAMI_DEBUG
+      value: {{ ternary "true" "false" (or .Values.psqlImage.debug .Values.diagnosticMode.enabled) | quote }}
+    - name: DATABASE_HOST
+      value: {{ include "nessie.database.host" . | quote }}
+    - name: DATABASE_PORT_NUMBER
+      value: {{ include "nessie.database.port" . | quote }}
+    {{- if not .Values.usePasswordFile }}
+    - name: DATABASE_PASSWORD
+      valueFrom:
+        secretKeyRef:
+          name: {{ include "nessie.database.secretName" . }}
+          key: {{ include "nessie.database.passwordKey" . }}
+    {{- end }}
+    - name: DATABASE_USER
+      value: {{ include "nessie.database.user" . | quote }}
+    - name: DATABASE_NAME
+      value: {{ include "nessie.database.name" . | quote }}
+  volumeMounts:
+    - name: empty-dir
+      mountPath: /tmp
+      subPath: tmp-dir
+    {{- if .Values.usePasswordFile }}
+    - name: database-password
+      mountPath: /bitnami/nessie/secrets/database
+    {{- end }}
+{{- end -}}
+
+{{/*
+Compile all warnings into a single message.
+*/}}
+{{- define "nessie.validateValues" -}}
+{{- $messages := list -}}
+{{- $messages := append $messages (include "nessie.validateValues.postgresql" .) -}}
```

Comment:
Poor list, improve it by adding extra validation (e.g. can you have N replicas with persistence enabled? Is it mandatory to use RWX PVCs?).

Reply:
The use of ReadWriteMany is debatable. You may want to try multi-concurrency in one node and that would work with a ReadWriteOnly. 

I added an extra validation for the extraVolumes